src/main/java/org/candlepin/model/UserCurator.java

```diff
@@ -32,18 +34,39 @@ public UserCurator() {
     }
 
     public User findByLogin(String login) {
-        return (User) createSecureCriteria()
-            .add(Restrictions.eq("username", login))
-            .uniqueResult();
+        EntityManager em = getEntityManager();
+        CriteriaBuilder cb = em.getCriteriaBuilder();
+        CriteriaQuery<User> cq = cb.createQuery(User.class);
+        Root<User> root = cq.from(User.class);
+
+        Predicate usernamePredicate = cb.equal(root.get("username"), login);
+        Predicate securityPredicate = getSecurityPredicate(User.class, cb, root);
+
+        if (securityPredicate != null) {
+            cq.where(cb.and(usernamePredicate, securityPredicate));
```

Comment:
FYI: It's not necessary to make the conjunction for a situation like this. The `.where` method is variadic and multiple predicates will be applied as a conjunction, so the following code is functionally identical to this line:

```
cq.where(usernamePredicate, securityPredicate);
```

Reply:
good to know. but isn't it better if the and is there? if someone goes through the code after a while, they might not remember that there is an implicit and.

Reply:
Depends on how you read it and whether or not the reader understands that as well. It's functionally identical and probably does exactly what you did under the hood anyway, so if you'd rather be explicit in such cases, that's fine, too.